javers-core/src/main/java/org/javers/core/JaversBuilder.java

```diff
@@ -477,6 +479,37 @@ public JaversBuilder withObjectAccessHook(ObjectAccessHook objectAccessHook) {
         return this;
     }
 
+
+    /**
+     * Registers a custom comparator for a given Value type
+     * (type of a property owned by Entity or ValueObject).
+     *
+     * Comparator which produces {@link ValueChange} should be registered by this method, it causes that comparator will
+     * be used during comparing {@link List}, {@link Set} and {@link java.util.Map}
+     *
+     * <br/><br/>
+     *
+     * Custom comparators are used by diff algorithm to calculate property-to-property diff.
+     * <br/><br/>
+     *
+     * Comparator has to calculate and return a {@link ValueChange}.
+     * <br/><br/>
+     *
+     * Internally, given type is mapped as {@link CustomType}.
+     * <br/><br/>
+     *
+     * Custom types are serialized to JSON using Gson defaults.
+     *
+     * @param <T> custom type
+     * @param customType class literal to define a custom type
+     * @see CustomPropertyComparator
+     */
+    public <T> JaversBuilder registerValueChangeCustomComparator(CustomPropertyComparator<T, ValueChange> comparator, Class<T> customType){
```

Comment:
The new API is unclear for me. Why Custom Comparator for Collections of Values returns `ValueChange`?
When we compare Collections we produce  `ContainerChange` with list of entry changes (add, remove, etc).

Second, does it affect *normal* comparing of Values when they are simple properties and not elements of a Collection?

I think, that we should somehow extend the `CustomPropertyComparator` interface to support Collections,
and let user to register one CustomPropertyComparator per one Custom type